app/controllers/lessons_controller.rb

```diff
@@ -32,12 +32,17 @@ def confirm
    
   def create
     @lesson = Lesson.new(lesson_params)
```

Comment:
Another way to do this is `current_user.lessons.build(lesson_params)`. This will automatically associate the user to the lesson and you can remove `whitelisted.merge(teacher_id: current_user.id.to_i)` from `#lesson_params`.